### DISCUSSION TITLE
LOCAL model can't read Long Notes

### DISCUSSION THREAD

**Discussion by fabiensc0ville** [logancyang/obsidian-copilot#562] [Q&A]
**Describe how to reproduce**
Both _Chat_ and _Long Note QA_ work great with LOCAL OLLAMA. However when I try local model on a **long** note, it ignores it, and comes back to the default prompt about Obsidian and AI

**Expected behavior**
Answer questions about the active note

**Screenshots**
| CHAT MODE  | LONG NOTE QA |
| ------------- | ------------- |
| <img width="400" alt="Chat mode" src="https://github.com/user-attachments/assets/fa10e0aa-a3a9-4bd1-b61f-81d9ba19250d"> |<img width="400" alt="LONG QA" src="https://github.com/user-attachments/assets/d2ee5982-1260-45f8-8717-a677d637d3b7">  |

**Additional context**
I believe the difference is the note length because it is the obvious one (it has 1 000 000 char and 1.2MB file size!), but I am not sure about it

**Comment by kiradzS** [comment]
Consumer-grade graphics cards cannot support the complete context token of the local model

**Comment by logancyang** [comment]
@fabiensc0ville have you checked the doc and set the context window explicitly in Ollama? https://github.com/logancyang/obsidian-copilot/blob/master/local_copilot.md#ollama

Also you didn't mention which model you were using, how long is its context window?
